lib/sycamore/sycamore/utils/cache.py

```diff
@@ -11,23 +11,44 @@
 BLOCK_SIZE = 1048576  # 1 MiB
 
 
+class HashWrapper:
```

Comment:
I'm not sure what value is added by this wrapper.  Maybe we could get by with a string constant set to "sha256" and callers can just call hashlib.new(STRING_CONSTANT) themselves.

Reply:
It helps passing around the hash ctx as Python/mypy/IDE does not like me accessing _Hash from hashlib

Reply:
Got it.  Maybe a comment on the class saying that would help.